src/core/Akka/Util/ByteString.cs

```diff
@@ -539,6 +539,7 @@ public ByteString Concat(ByteString other)
         /// <returns>TBD</returns>
         public int CopyTo(byte[] buffer, int index, int count)
         {
+            if(buffer.Length == 0 && count == 0) return 0; // edge case for no-copy
```

Comment:
Added the same bounds check to all three implementations of the `CopyTo` method.